chain-signatures/node/src/http_client.rs

```diff
@@ -177,6 +187,22 @@ impl MessageQueue {
                 let account_id = &info.account_id;
                 let number_of_messages = encrypted_partition.len() as f64;
 
+                let mut presignature_msg_creation_time: HashMap<u64, u64> = HashMap::new();
+                for (_, message, _) in &msgs {
+                    if let MpcMessage::Presignature(PresignatureMessage { id, timestamp, .. }) =
+                        message
+                    {
+                        presignature_msg_creation_time
+                            .entry(*id)
+                            .and_modify(|ts| {
+                                if *timestamp > *ts {
+                                    *ts = *timestamp;
+                                }
+                            })
+                            .or_insert(*timestamp);
+                    }
+                }
+
```

Comment:
I am not sure I'm following the intention here exactly.
I think this calculates the minimum timestamp of messages for each id.

But why? Is this to have a single delay for a batch of messages sent at once? Maybe a short comment could help to make it easier to understand.